c2rust-refactor/src/context.rs

```diff
@@ -733,6 +965,10 @@ impl<'hir> HirMap<'hir> {
             return *id;
         }
 
+        // TODO: consult `context_to_node_id_map` using the structural fingerprint recorded by
+        // `SpanToHirMapper`.  Falling straight back to `opt_local_def_id` still conflates siblings
+        // that share a span (e.g. derive helpers), so HIR -> NodeId lookups remain lossy until we
+        // thread the same context key through this path.
```

Comment:
As mentioned in meeting, it would be nice to clarify what uses `hir_to_node_id` and where we might see the consequences of not using the structural fingerprint here.